ui/perfherder/graphs/GraphsContainer.jsx

```diff
@@ -213,6 +213,17 @@ class GraphsContainer extends React.Component {
     return numberFormat.format(tick);
   };
 
+  checkDate = x => {
+    let isVisible = false;
+    // eslint-disable-next-line no-return-assign
+    this.props.testData.map(item =>
+      item.visible ? (isVisible = true) : isVisible,
+    );
+    return isVisible
+      ? moment.utc(x).format('MMM DD')
+      : moment.utc(new Date()).format('MMM DD');
```

Comment:
This was a good attempt :)

We only disable a few select eslint rules but usually for specific reasons (like a poorly written rule). In this case, it's a hint that you are doing something that isn't best practice (`isVisible = true`). I suggest reading the rule for future reference :)

So this approach doesn't quite make sense - we don't want to return the date based on the last value `isVisible` is set to - we want to make a quick determination about whether not we can consider `testData` to be empty or to not contain items that are visible. `filter` is a better method than `map` to use here. Sometimes it helps to write a function in pseudo-code to make sure the logic is correct.

Also I think if you do `moment().utc().format('MMM DD')` that achieves the same thing as `moment.utc(new Date()).format('MMM DD')`.

Reply:
There are merge conflicts so you will need to rebase onto the latest master (don't do pull and merge).

Reply:
@sarah-clements Thank you for your feedback, I see my mistakes now and tried to go through the prob in a better way, I think.

What I did is:
```
 checkDate = x => {
    let newArr = [];
    newArr = this.props.testData.filter(
      item => item.visible === true && item.data.length > 0,
    );

    return newArr.length > 0
      ? moment.utc(x).format('MMM DD')
      : moment.utc().format('MMM DD');
  };
```

- When we check cards, and visible prop is true, but the card data is empty, we show the current date. 
`Question: Do you want to show the current date in selected cards, with no data inside it? If we don't want to show the current date, but check only the visible prop is true, then the 01 Jan will be shown.`
- Do we have to check for the visible and data if they are true? 
- The solution I provide checks the visible prop and the data length for cards that are selected -> save items that have these two things, if we have data in the new array then show the date from the data, if not, show the current date.
- Please let me know if you have a better approach, thank you.
- I will not push the code yet.

Reply:
This looks better. To answer your question, we definitely don't want to show the 01 Jan date at all. I'm assuming you tested this approach using the link in this [comment](https://github.com/mozilla/treeherder/pull/5440#issuecomment-538133224)?

A few nits:
- You should declare `newArr` once (instead of declaring it with an empty array and reassigning it) because `filter` will return an empty array if there aren't any items that meet the condition: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter
- Rename `newArr` to something meaningful like `graphData` or `data`
- I think you need to call `moment().utc().format('MMM DD')` - to get today's date. You can check the momentjs docs to be certain

Reply:
- I tested this approach with the link in that comment.
- We don't show the 01 Jan then, the current date will be shown when visible is true but there is no data.
- Thank you for your feedback on not declaring the `newArr`
I created a const and attach to it the calculation.
```
  checkDate = x => {
    const graphData = this.props.testData.filter(
      item => item.visible === true && item.data.length > 0,
    );

    return graphData.length > 0
      ? moment.utc(x).format('MMM DD')
      : moment().format('MMM DD');
  };
```
- Related to the moment library:
	* If we want to show the users local time from moment library, then we can use the `moment().format('MMM DD')` - While for the UTC - the docs says:  all display methods will display in UTC time instead of local time.
	* The difference:
		``` 
		moment().format();     // 2013-02-04T10:35:24-08:00
		moment.utc().format(); // 2013-02-04T18:35:24+00:00 
		```
	* Which one to use? I think using either option would work because we are not showing the hour, what do you think?

Reply:
ah ok, we should only being showing utc date - not localized - so let's go with that format (it does matter because at some point local date for PST (for example) could be 2013-02-04 but be 2013-02-05 UTC).

Reply:
@sarah-clements Hi, can you give me a hint related to the failed `continuous-integration/travis-ci/`

Reply:
These errors are unrelated to what you've done. We need to revert someone else's commit that caused issues. I'll let you know when it's fixed.

Reply:
This has been fixed now.